core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -3142,37 +3124,26 @@ select json_extract(json,path)::double -> select json_extract(json,path,double)
        select json_extract(json,path)::uuid -> select json_extract(json,path)::uuid
 
        Notes:
-        - varchar cast it rewritten in a special way, e.g. removed
+        - varchar cast it rewritten in a special way, i.e., removed
         - subset of types is handled more efficiently in the 3-arg function
-        - the remaining type casts are not rewritten, e.g. left as is
+        - the remaining type casts are not rewritten, i.e., left as is
      */
 
     private ExpressionNode rewriteKnownStatements(
             ExpressionNode parent,
             @Nullable LowerCaseCharSequenceObjHashMap<ExpressionNode> decls,
             @Nullable CharSequence exprTargetVariableName
     ) throws SqlException {
-        return
-                rewriteDeclaredVariables(
-                        rewriteJsonExtractCast(
-                                rewritePgCast(
-                                        rewriteConcat(
-                                                rewriteCase(
-                                                        rewriteCount(
-                                                                parent
-                                                        )
-                                                )
-                                        )
-                                )
-                        ), decls, exprTargetVariableName);
-    }
-
-    private ExpressionNode rewritePgCast(ExpressionNode parent) throws SqlException {
-        traversalAlgo.traverse(parent, rewritePgCast0Ref);
-        return parent;
+        traversalAlgo.traverse(parent, rewriteCountRef);
+        traversalAlgo.traverse(parent, rewriteQueryTraceFunctionRef);
```

Comment:
why this change was needed? this traverses tree multiple times instead of once as before.

Reply:
It always did the same thing, this came about by pure inlining.